comms/tests/connection/peer_connection.rs

```diff
@@ -217,7 +214,7 @@ fn connection_wait_connect_failed() {
 
 #[test]
 fn connection_pause_resume() {
-    let addr = find_available_tcp_net_address("127.0.0.1").unwrap();
+    let addr = NetAddress::from_str("127.0.0.1:9873").unwrap();
```

Comment:
Think a hard-coded port number could cause issues, if we rely on devs to use a unique port for their tests - the reason that the connection manager port allocations had a race condition was because there were multiple instances of connection manager, the test helper `find_available_tcp_net_address` port tracking is static and guarantees that you'll be able to "book" a port for the test. I don't think it was the cause of any flakiness.

Reply:
I'm worried that calling `bind` to see if a port is free and then giving it to zmq was also a source of flakiness -- if the OS holds onto the port for any length of time before zmq wants to connect...